### PR TITLE
Track overflow:scroll stacking contexts with ScrollRootId instead of StackingContextId

### PR DESCRIPTION
**Pull request by mrobinson**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because this PR should not change behavior.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13957)

### CODE REVIEW

**Comment by mrobinson** [comment]
@glennw or @pcwalton r?

**Comment by jdm** [comment]
That is not the actual log that shows the failure; I filed https://github.com/servo/saltfs/issues/530 about this problem.

File: components/gfx/display_list/mod.rs [hunk 1]
```diff
@@ -214,10 +214,12 @@ impl DisplayList {
         // We don't perform this adjustment on the root stacking context because
         // the DOM-side code has already translated the point for us (e.g. in
         // `Window::hit_test_query()`) by now.
-        if !is_fixed && stacking_context.id != StackingContextId::root() {
-            if let Some(scroll_offset) = scroll_offsets.get(&stacking_context.id) {
-                translated_point.x -= Au::from_f32_px(scroll_offset.x);
-                translated_point.y -= Au::from_f32_px(scroll_offset.y);
+        if stacking_context.id != StackingContextId::root() {
```
Comment (glennw) [inline components/gfx/display_list/mod.rs]: Shouldn't this still check !is_fixed?

**Review by glennw — COMMENTED** [review]
One question, r=me once that's resolved.

File: components/gfx/display_list/mod.rs (hunk 1 above)
Comment (mrobinson) [inline components/gfx/display_list/mod.rs]: Okay. Thanks for pointing that out! I've fixed it in the latest version of the branch.

**Pull request merged**